exporter/otlp-http/lib/opentelemetry/exporter/otlp/http/exporter.rb

```diff
@@ -0,0 +1,303 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/common'
+require 'opentelemetry/exporter/otlp/common'
+require 'opentelemetry/sdk'
+require 'net/http'
+require 'csv'
+require 'zlib'
+
+require 'google/rpc/status_pb'
+
+module OpenTelemetry
+  module Exporter
+    module OTLP
+      module HTTP
+        # An OpenTelemetry trace exporter that sends spans over HTTP as Protobuf encoded OTLP ExportTraceServiceRequests.
+        class Exporter # rubocop:disable Metrics/ClassLength
+          SUCCESS = OpenTelemetry::SDK::Trace::Export::SUCCESS
+          FAILURE = OpenTelemetry::SDK::Trace::Export::FAILURE
+          private_constant(:SUCCESS, :FAILURE)
+
+          # Default timeouts in seconds.
+          KEEP_ALIVE_TIMEOUT = 30
+          RETRY_COUNT = 5
+          WRITE_TIMEOUT_SUPPORTED = Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.6')
+          private_constant(:KEEP_ALIVE_TIMEOUT, :RETRY_COUNT, :WRITE_TIMEOUT_SUPPORTED)
+
+          ERROR_MESSAGE_INVALID_HEADERS = 'headers must be a String with comma-separated URL Encoded UTF-8 k=v pairs or a Hash'
+          private_constant(:ERROR_MESSAGE_INVALID_HEADERS)
+
+          def self.ssl_verify_mode
```

Comment:
Was this method always public? Why?

Reply:
It appears to be currently public, https://github.com/open-telemetry/opentelemetry-ruby/blob/c521837879e38629b2facb8d01bc19cc784dbfc3/exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb#L38-L46

Not sure why.

Reply:
It should be private IMO - there's no reason I can see in the original PR for making it public. There's talk of extracting it to some common place for various exporters to use, but that hasn't been done. Let's make it private for now.